File: 백준/Bronze/2920. 음계/음계.cc
#include<iostream>
#include<cstdio>
#define LEN 8
enum {ASC=0, DESC, MIXED};
using namespace std;
 
int main(void){
    int result=0;
    int arr[LEN];
 
    for(int i=0; i<LEN; i++){
        scanf("%d", &arr[i]);
    }
 
    //초기 값 정해주기
    if(arr[1]-arr[0] == 1)result = ASC;
    else if(arr[1]-arr[0] == -1)result = DESC;
    else result = MIXED;
 
    //이전 값과 같은지 아닌지
    for(int i=1; i<LEN-1; i++){
        if(arr[i+1]-arr[i] == 1 && result == ASC) continue;
        if(arr[i+1]-arr[i] == -1 && result == DESC) continue;
 
        result = MIXED;
        break;
    }
 
    
    if(result == ASC) cout << "ascending";
    else if(result == DESC) cout << "descending";
    else cout << "mixed";
 
    return 0;
}

File: 백준/Bronze/2920. 음계/음계.java
import java.util.*;
import java.io.*;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static StringTokenizer st;

    public static void main(String[] args) throws Exception {
        int[] arr = new int[8];
        st = new StringTokenizer(br.readLine());

        for (int i = 0; i < 8; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        int status = -1;
        for (int i = 0; i < 7; i++) {
            if (arr[i] < arr[i + 1] && status != 1) {
                status = 0;
            } else if (arr[i] > arr[i + 1] && status != 0)
                status = 1;
            else {
                status = 2;
                break;
            }
        }
        switch (status) {
            case 0:
                bw.write("ascending");
                break;
            case 1:
                bw.write("descending");
                break;
            case 2:
                bw.write("mixed");
                break;
        }
        bw.close();
    }
}